Adds binary recursive stein algorithm to the GCD class

kayoslab opened the pull request:
<!-- Thanks for contributing to the SAC! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] -->

### Checklist

- [x] I've looked at the [contribution guidelines](https://github.com/raywenderlich/swift-algorithm-club/blob/master/.github/CONTRIBUTING.md).
- [x] This pull request is complete and ready for review.

### Description

<!-- In a short paragraph, describe the PR --> 
* Three different approaches for the GCD to be calculated
* Improves the LCM by passing the used GCD algorithm into the function.
* Find an easy GCD function if possible.
* Adopts the changes in the appropriate README file.

richard-ash commented on GCD/README.markdown:
```diff
@@ -88,9 +88,9 @@ The `max()` and `min()` at the top of the function make sure we always divide th
+The least common multiple of two numbers `a` and `b` is the smallest positive integer that is a multiple of both. In other words, the LCM is evenly divisible by `a` and `b`. The example implementation of the LCM takes two numbers and a specification which GCD algorithm is used.
 
-For example: `lcm(2, 3) = 6` because 6 can be divided by 2 and also by 3.
+For example: `lcm(2, 3, using: gcdIterativeEuklid) = 6` because 6 can be divided by 2 and also by 3.
```
Keep this as `lcm(2, 3) = 6`

richard-ash commented on GCD/GCD.swift:
```diff
@@ -1,34 +1,124 @@
+ an unsigned integer
+ */
+func lcm(_ m: Int, _ n: Int, using gcdAlgorithm: (Int, Int) -> (Int)) throws -> Int {
+    guard (m & n) != 0 else { throw LCMError.divisionByZero }
```
nit: You don't need parentheses around `m & n`

kayoslab commented:
@richard-ash Thank you for the valid comments. I updated my pull request according to that. The example code is now moved into the playground's `Sources` folder and the README is extended with details regarding all three algorithms and a short discussion.

richard-ash commented:
Awesome thanks so much @kayoslab. Changes look 💯
